### PR TITLE
W-12594415: HTTP Connector POST 303 Redirect passing body into the GET Request

### PR DESCRIPTION
**Pull request by marianorean**

### CODE REVIEW

File: src/main/java/org/mule/extension/http/internal/request/HttpClientReflection.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import org.mule.extension.http.api.request.HttpSendBodyMode;
+import org.mule.runtime.http.api.client.HttpClient;
+import org.mule.runtime.http.api.client.auth.HttpAuthentication;
+import org.mule.runtime.http.api.domain.message.request.HttpRequest;
+import org.mule.runtime.http.api.domain.message.response.HttpResponse;
+
+import java.lang.reflect.Method;
+import java.util.concurrent.CompletableFuture;
+
+public class HttpClientReflection {
```
Comment (eze210) [inline src/main/java/org/mule/extension/http/internal/request/HttpClientReflection.java:20]: Add javadoc

File: src/main/java/org/mule/extension/http/internal/request/HttpClientReflection.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+  public static final String HTTP_REQUEST_OPTIONS_CLASS_NAME =
+      "org.mule.runtime.http.api.client.HttpRequestOptions";
+
+  private Object requestOptionsBuilder;
+  Method sendAsyncMethod;
```
Comment (eze210) [inline src/main/java/org/mule/extension/http/internal/request/HttpClientReflection.java:26]: private?

File: src/main/java/org/mule/extension/http/internal/request/HttpClientReflection.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  Method buildMethod;
+  Method responseTimeoutMethod;
+  Method followsRedirectMethod;
+  Method authenticationMethod;
+  Method sendBodyAlwaysMethod;
+
+  boolean loaded = true;
+
+  public HttpClientReflection(HttpClient client) {
```
Comment (eze210) [inline src/main/java/org/mule/extension/http/internal/request/HttpClientReflection.java:35]: This way you're doing the reflection thing once per client. Maybe you can have only one instance of this class and receive the client in the sendAsync

**Review by eze210 — CHANGES_REQUESTED** [review]

File: src/main/java/org/mule/extension/http/internal/request/HttpClientReflection.java (hunk 1 above)
Comment (marianorean) [reply src/main/java/org/mule/extension/http/internal/request/HttpClientReflection.java:23]: done

**Review by eze210 — CHANGES_REQUESTED** [review]
The change is ok. Only missing thing now is an MUnit test

File: src/main/java/org/mule/extension/http/internal/request/HttpClientReflection.java (hunk 2 above)
Comment (marianorean) [reply src/main/java/org/mule/extension/http/internal/request/HttpClientReflection.java]: done

File: src/main/java/org/mule/extension/http/internal/request/HttpClientReflection.java (hunk 3 above)
Comment (marianorean) [reply src/main/java/org/mule/extension/http/internal/request/HttpClientReflection.java]: done

**Review by eze210 — APPROVED** [review]

**Pull request merged**